rllib/core/learner/torch/torch_learner.py

```diff
@@ -257,15 +262,57 @@ def apply_gradients(self, gradients_dict: ParamDict) -> None:
-                        for scheduler in self._lr_schedulers[module_id][optimizer_name]:
-                            scheduler.step()
-
-                    # If the module uses learning rate schedulers, step them here.
-                    if module_id in self._lr_schedulers:
+    @OverrideToImplementCustomLogic_CallToSuperRecommended
+    @override(Learner)
+    def after_gradient_based_update(self, *, timesteps: Dict[str, Any]) -> None:
+        """Called after gradient-based updates are completed.
+
+        Should be overridden to implement custom cleanup-, logging-, or non-gradient-
+        based Learner/RLModule update logic after(!) gradient-based updates have been
+        completed.
+
+        Note, for `framework="torch"` users can register
+        `torch.optim.lr_scheduler.LRScheduler` via
+        `AlgorithmConfig._torch_lr_scheduler_classes`. These schedulers need to be
+        stepped here after gradient updates and reported.
+
+        Args:
+            timesteps: Timesteps dict, which must have the key
+                `NUM_ENV_STEPS_SAMPLED_LIFETIME`.
+                # TODO (sven): Make this a more formal structure with its own type.
+        """
+        # If we have `torch.optim.lr_scheduler.LRScheduler` we need to step them here
+        # and report learning rates.
+        if self._lr_schedulers:
+            # Only update this optimizer's lr, if a scheduler has been registered
+            # along with it.
+            for module_id, optimizer_names in self._module_optimizers.items():
+                for optimizer_name in optimizer_names:
+                    # If learning rate schedulers are provided step them here. Note,
+                    # stepping them in `TorchLearner.apply_gradients` updates the
+                    # learning rates during minibatch updates; we want to update
+                    # between whole batch updates.
+                    if (
+                        module_id in self._lr_schedulers
+                        and optimizer_name in self._lr_schedulers[module_id]
+                    ):
                         for scheduler in self._lr_schedulers[module_id][optimizer_name]:
                             scheduler.step()
+                    optimizer = self.get_optimizer(module_id, optimizer_name)
+                    self.metrics.log_value(
+                        # Cut out the module ID from the beginning since it's already
+                        # part of the key sequence: (ModuleID, "[optim name]_lr").
+                        key=(
+                            module_id,
+                            f"{optimizer_name[len(module_id) + 1:]}_{LR_KEY}",
+                        ),
+                        value=convert_to_numpy(self._get_optimizer_lr(optimizer)),
+                        window=1,
+                    )
+        # Otherwise call the `super()`'s method to update RLlib's learning rate
+        # schedules.
+        else:
```

Comment:
super nit: Can we flip this if-block and move the default RLlib case to the top? Then we don't even need an else block anymore and can save the long indentation for the rest of this method.

Reply:
Will do. Makes totally sense.